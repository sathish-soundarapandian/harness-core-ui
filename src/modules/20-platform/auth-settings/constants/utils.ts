/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { windowLocationUrlPartBeforeHash } from 'framework/utils/WindowLocation'

export const getSamlEndpoint = (accountId: string): string => {
  let url = `${windowLocationUrlPartBeforeHash()?.replace('/ng/', '')}`

  if (window.apiUrl) {
    if (window.apiUrl.startsWith('/')) {
      url = `${url}${window.apiUrl}`
    } else {
      url = window.apiUrl
    }
  }

  return `${url}/api/users/saml-login?accountId=${accountId}`
}
