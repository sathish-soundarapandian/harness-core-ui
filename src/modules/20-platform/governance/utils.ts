/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { GovernanceMetadata } from 'services/cd-ng'

interface DoesGovernanceMetaDataHasErrorOrWarningRtn {
  governanceMetaDataHasError: boolean
  governanceMetaDataHasWarning: boolean
}
export const doesGovernanceHasErrorOrWarning = (
  governanceMetadata: GovernanceMetadata | undefined
): DoesGovernanceMetaDataHasErrorOrWarningRtn => {
  let governanceMetaDataHasWarning = false
  let governanceMetaDataHasError = false
  if (governanceMetadata && governanceMetadata?.status === 'error') {
    governanceMetaDataHasError = true
  }
  if (governanceMetadata && governanceMetadata?.status === 'warning') {
    governanceMetaDataHasWarning = true
  }
  return { governanceMetaDataHasError, governanceMetaDataHasWarning }
}
