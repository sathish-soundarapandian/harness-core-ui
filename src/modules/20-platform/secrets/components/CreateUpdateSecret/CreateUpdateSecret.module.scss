/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

.hoverMsg {
  display: block;
}
.hoverMsgTarget {
  display: block !important;
}
.referenceSecret {
  width: 80%;
  margin-bottom: 0px;
}
.refernceSecretLayout {
  margin-bottom: var(--spacing-medium) !important;
}
.testBtn {
  align-self: flex-start;
  margin-top: var(--spacing-7) !important;
}
.expiresOnLabel {
  color: var(--grey-600);
  font-size: var(--form-input-font-size);
  font-weight: 500;
}
