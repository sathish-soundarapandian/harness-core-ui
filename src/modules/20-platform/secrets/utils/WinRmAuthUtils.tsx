/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { WinRmConfigFormData } from '@secrets/modals/CreateWinRmCredModal/views/StepAuthentication'
import {
  KerberosWinRmConfigDTO,
  NTLMConfigDTO,
  TGTKeyTabFilePathSpecDTO,
  TGTPasswordSpecDTO,
  WinRmCredentialsSpecDTO,
  getSecretV2Promise,
  KerberosConfigDTO,
  SecretDTOV2
} from 'services/cd-ng'

import type { SecretReference } from '@secrets/components/CreateOrSelectSecret/CreateOrSelectSecret'

function buildKerberosConfig(data: WinRmConfigFormData): KerberosWinRmConfigDTO {
  switch (data.tgtGenerationMethod) {
    case 'KeyTabFilePath':
      return {
        principal: data.principal,
        realm: data.realm,
        tgtGenerationMethod: data.tgtGenerationMethod,
        useSSL: data.useSSL,
        useNoProfile: data.useNoProfile,
        skipCertChecks: data.skipCertChecks,
        spec: {
          keyPath: data.keyPath
        } as TGTKeyTabFilePathSpecDTO
      } as KerberosWinRmConfigDTO
    case 'Password':
      return {
        principal: data.principal,
        realm: data.realm,
        tgtGenerationMethod: data.tgtGenerationMethod,
        useSSL: data.useSSL,
        useNoProfile: data.useNoProfile,
        skipCertChecks: data.skipCertChecks,
        spec: {
          password: data.password?.referenceString
        } as TGTPasswordSpecDTO
      } as KerberosWinRmConfigDTO
    default:
      return {
        principal: data.principal,
        realm: data.realm,
        useSSL: data.useSSL,
        useNoProfile: data.useNoProfile,
        skipCertChecks: data.skipCertChecks
      }
  }
}

export function buildAuthConfig(data: WinRmConfigFormData): NTLMConfigDTO | KerberosWinRmConfigDTO {
  switch (data.authScheme) {
    case 'NTLM':
      return {
        username: data.username,
        password: data.password?.referenceString,
        domain: data.domain,
        useNoProfile: data.useNoProfile,
        useSSL: data.useSSL,
        skipCertChecks: data.skipCertChecks
      } as NTLMConfigDTO
    case 'Kerberos':
      return buildKerberosConfig(data)
  }
}

export const getSecretReferencesForWinRm = async (
  secret: SecretDTOV2,
  pathParams: {
    accountIdentifier: string
    orgIdentifier?: string
    projectIdentifier?: string
  }
): Promise<{
  keySecret?: SecretReference
  passwordSecret?: SecretReference
  encryptedPassphraseSecret?: SecretReference
}> => {
  let password, passwordSecret
  const secretSpec = secret.spec as WinRmCredentialsSpecDTO
  const authScheme = secretSpec.auth.type
  if (authScheme === 'NTLM') {
    const ntlmConfig = secretSpec.auth.spec as NTLMConfigDTO
    password = ntlmConfig?.password
  } else if (authScheme === 'Kerberos') {
    const kerberosConfig = secretSpec.auth.spec as KerberosConfigDTO
    if (kerberosConfig.tgtGenerationMethod === 'Password') {
      password = kerberosConfig.spec?.password
    }
  }

  if (password) {
    const secretId = password.indexOf('.') < 0 ? password : password.split('.')[1]
    const data = await getSecretV2Promise({
      identifier: secretId,
      queryParams: pathParams
    })
    passwordSecret = {
      ...data.data?.secret,
      referenceString: password
    } as SecretReference
  }

  return {
    passwordSecret
  }
}
