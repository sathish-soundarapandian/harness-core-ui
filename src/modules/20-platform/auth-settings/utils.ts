/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { get } from 'lodash-es'
import type { CreateIpAllowlistConfigProps, UpdateIpAllowlistConfigProps } from '@harnessio/react-ng-manager-client'

import type { IIPAllowlistForm } from '@auth-settings/interfaces/IPAllowlistInterface'
import { SourceType } from '@auth-settings/interfaces/IPAllowlistInterface'
import { CurrencyType } from '@common/constants/SubscriptionTypes'

export function getAmountInCurrency(currency: CurrencyType, amount: number): string {
  switch (currency) {
    case CurrencyType.USD: {
      return amount.toLocaleString('en-US', {
        style: 'currency',
        currency: 'USD'
      })
    }
  }

  return ''
}

// dividing the actual amount by 100 as amount numbers are in cents
export function getDollarAmount(amount?: number, isYearly?: boolean): number {
  if (isYearly) {
    return (amount || 0) / 1200
  }
  return (amount || 0) / 100
}

export const getErrorMessage = (error: any): string =>
  get(error, 'data.error', get(error, 'data.message', error?.message))

export const buildCreateIPAllowlistPayload = (data: IIPAllowlistForm): CreateIpAllowlistConfigProps => {
  const allowSourceType = []
  if (data.allowSourceTypeUI) {
    allowSourceType.push(SourceType.UI)
  }
  if (data.allowSourceTypeAPI) {
    allowSourceType.push(SourceType.API)
  }
  return {
    body: {
      ip_allowlist_config: {
        allowed_source_type: allowSourceType,
        description: data?.description,
        enabled: false, // always false while creating
        identifier: data.identifier,
        ip_address: data.ipAddress,
        name: data.name,
        tags: data?.tags
      }
    }
  }
}

export const buildUpdateIPAllowlistPayload = (
  data: IIPAllowlistForm,
  overrideFields?: any
): UpdateIpAllowlistConfigProps => {
  const allowSourceType = []
  if (data.allowSourceTypeUI) {
    allowSourceType.push(SourceType.UI)
  }
  if (data.allowSourceTypeAPI) {
    allowSourceType.push(SourceType.API)
  }
  return {
    'ip-config-identifier': data.identifier,
    body: {
      ip_allowlist_config: {
        allowed_source_type: allowSourceType,
        description: data?.description,
        enabled: data?.enabled,
        identifier: data.identifier,
        ip_address: data.ipAddress,
        name: data.name,
        tags: data?.tags,
        ...overrideFields
      }
    }
  }
}
