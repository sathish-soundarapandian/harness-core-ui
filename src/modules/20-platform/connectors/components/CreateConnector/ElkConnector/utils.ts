/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { ElkAuthType } from '@connectors/constants'
import type { FormData } from '@connectors/interfaces/ConnectorInterface'
import type { ConnectorConfigDTO } from 'services/cd-ng'

type InitializeElkConnectorArgs = {
  prevStepData?: ConnectorConfigDTO
  accountId: string
  projectIdentifier?: string
  orgIdentifier?: string
}

export function initializeElkConnector({
  prevStepData,
  accountId,
  projectIdentifier,
  orgIdentifier
}: InitializeElkConnectorArgs): FormData {
  const defaultObj = {
    url: '',
    accountId,
    username: '',
    authType: ElkAuthType.USERNAME_PASSWORD,
    password: undefined,
    projectIdentifier,
    orgIdentifier
  }

  if (!prevStepData) {
    return defaultObj
  }
  const { spec, ...prevData } = prevStepData
  return {
    ...defaultObj,
    url: prevData?.url || spec?.controllerUrl || spec?.url || '',
    password: prevData?.password || spec?.passwordRef,
    clientSecretRef: prevData?.clientSecretRef || spec?.clientSecretRef,
    authType: prevData?.authType || spec?.authType || ElkAuthType.USERNAME_PASSWORD,
    username: prevData?.username || spec?.username || '',
    clientId: prevData?.clientId || spec?.clientId,
    apiKeyId: prevData?.apiKeyId || spec?.apiKeyId,
    apiKeyRef: prevData?.apiKeyRef || spec?.apiKeyRef
  }
}
