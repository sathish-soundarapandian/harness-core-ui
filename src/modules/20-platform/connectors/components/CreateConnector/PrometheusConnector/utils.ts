/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { ValueType } from '@secrets/components/TextReference/TextReference'
import { setSecretField } from '@secrets/utils/SecretField'
import type { ConnectorConfigDTO, GetSecretV2QueryParams } from 'services/cd-ng'
import type { CustomHealthConnectorDTO, CustomHealthKeyAndValue } from 'services/cv'
import type {
  BaseCompFields,
  EncryptedEntity,
  PlainEntity
} from '../CustomHealthConnector/components/CustomHealthHeadersAndParams/CustomHealthHeadersAndParams.types'
import { DefaultHeadersInitialValues } from './CreatePrometheusConnector.constants'

async function transformSpecHeader(
  scopeQueryParams: GetSecretV2QueryParams,
  entity?: CustomHealthConnectorDTO['headers']
): Promise<BaseCompFields['headers']> {
  if (!entity?.length) {
    return DefaultHeadersInitialValues
  }

  const transformedEntities: BaseCompFields['headers'] = []
  const encryptedEntities = entity.filter(e => {
    if (!e.valueEncrypted) {
      transformedEntities.push({
        key: e.key,
        value: {
          textField: e.value,
          fieldType: ValueType.TEXT
        }
      } as PlainEntity)
    }
    return e.valueEncrypted
  })

  await Promise.all(
    encryptedEntities.map(encryptedEntity =>
      setSecretField(encryptedEntity.encryptedValueRef as string, scopeQueryParams)
    )
  )
    .then(secretObjs => {
      encryptedEntities.forEach((e, index) =>
        transformedEntities.push({
          key: e.key,
          value: {
            fieldType: ValueType.ENCRYPTED,
            secretField: secretObjs[index]
          }
        } as EncryptedEntity)
      )
    })
    .catch(() => ({}))

  return transformedEntities
}

export async function initializePrometheusConnectorWithStepData(
  prevStepData?: ConnectorConfigDTO | null,
  scopeQueryParams?: GetSecretV2QueryParams
): Promise<ConnectorConfigDTO | undefined> {
  if (!prevStepData) {
    return
  }

  const { spec, ...prevData } = prevStepData
  const updatedInitialValues = {
    ...spec,
    ...prevData
  }

  if (prevData?.url) {
    return {
      ...updatedInitialValues,
      url: prevData?.url,
      headers: prevData?.headers,
      username: prevData?.username,
      passwordRef: prevData?.passwordRef
    }
  }

  return {
    ...updatedInitialValues,
    url: spec.url,
    headers: await transformSpecHeader(scopeQueryParams as GetSecretV2QueryParams, spec.headers),
    username: spec?.username,
    passwordRef: await setSecretField(spec?.passwordRef, scopeQueryParams as GetSecretV2QueryParams)
  }
}

function getSpecHeaderAndEntity(entity: PlainEntity | EncryptedEntity): CustomHealthKeyAndValue | null {
  if (entity?.value?.fieldType === ValueType.ENCRYPTED && entity.key && entity.value.secretField?.referenceString) {
    return {
      valueEncrypted: true,
      encryptedValueRef: entity.value.secretField.referenceString,
      key: entity.key
    }
  } else if (entity?.value?.fieldType === ValueType.TEXT && entity.key && entity.value.textField) {
    return {
      valueEncrypted: false,
      value: entity.value.textField,
      key: entity.key
    }
  }

  return null
}

export function transformStepHeadersAndParamsForPayloadForPrometheus(headers: BaseCompFields['headers']): {
  headers: CustomHealthConnectorDTO['headers']
} {
  const apiHeaders: CustomHealthConnectorDTO['headers'] = []

  headers?.forEach(header => {
    const transformedHeader = getSpecHeaderAndEntity(header)
    if (transformedHeader) {
      apiHeaders.push(transformedHeader)
    }
  })

  return {
    headers: apiHeaders
  }
}
