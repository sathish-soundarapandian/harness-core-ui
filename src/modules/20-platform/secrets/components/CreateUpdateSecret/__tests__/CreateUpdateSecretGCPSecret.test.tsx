/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render, waitFor } from '@testing-library/react'

import { TestWrapper } from '@common/utils/testUtils'
import CreateUpdateSecret from '../CreateUpdateSecret'

import mockData from './listSecretManagersMock.json'
import { gcpSecretMock, gcpConnector } from './gcpSecretManagerMock'

const mockUpdateTextSecret = jest.fn()

jest.mock('services/cd-ng', () => ({
  usePutSecret: jest.fn().mockImplementation(() => ({ mutate: mockUpdateTextSecret })),
  usePostSecret: jest.fn().mockImplementation(() => ({ mutate: jest.fn() })),
  usePostSecretFileV2: jest.fn().mockImplementation(() => ({ mutate: jest.fn() })),
  usePutSecretFileV2: jest.fn().mockImplementation(() => ({ mutate: jest.fn() })),
  useGetGcpRegions: jest.fn().mockImplementation(() => ({
    mutate: jest.fn(),
    refetch: jest.fn(() =>
      Promise.resolve({
        status: 'SUCCESS',
        data: ['us-east1', 'us-east2', 'us-east4'],
        metaData: null,
        correlationId: '41bf7519-7c22-4543-93cb-fd38e5a4d95a'
      })
    ),
    data: null
  })),
  useGetSecretV2: jest.fn().mockImplementation(() => ({ refetch: jest.fn(), data: gcpSecretMock })),
  useGetConnectorList: () => {
    return {
      data: mockData,
      refetch: jest.fn()
    }
  },
  useGetConnector: () => {
    return {
      data: gcpConnector,
      refetch: jest.fn()
    }
  }
}))

describe('CreateUpdateSecret GCP Secret manager', () => {
  test('Text Secret in GCP', async () => {
    const { container, getByText } = render(
      <TestWrapper path="/account/:accountId/resources/secrets" pathParams={{ accountId: 'dummy' }}>
        <CreateUpdateSecret secret={gcpSecretMock as any} type={'SecretText'} />
      </TestWrapper>
    )

    await waitFor(() => expect(getByText('secrets.labelSecretName')).toBeTruthy())
    await waitFor(() => expect(getByText('secrets.secret.configureRegion')).toBeTruthy())
    expect(container).toMatchSnapshot()
  })
})
