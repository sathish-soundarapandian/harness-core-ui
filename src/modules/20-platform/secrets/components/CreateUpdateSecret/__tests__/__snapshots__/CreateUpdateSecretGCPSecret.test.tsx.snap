// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateUpdateSecret GCP Secret manager Text Secret in GCP 1`] = `
<div>
  <div
    class="OverlaySpinner--overlaySpinner"
  >
    <form
      action="#"
      class="FormikForm--main"
    >
      <div>
        <div
          class="bp3-form-group bp3-disabled"
          data-id="secretManagerIdentifier-0"
        >
          <label
            class="bp3-label"
            for="secretManagerIdentifier"
          >
            <span
              class="Tooltip--acenter"
              data-tooltip-id="createUpdateSecretForm_secretManagerIdentifier"
            >
              secrets.labelSecretsManager
            </span>
             
            <span
              class="bp3-text-muted"
            />
          </label>
          <div
            class="bp3-form-content"
          >
            <div
              class="bp3-popover-wrapper Select--main bp3-fill"
            >
              <div
                class="bp3-popover-target"
              >
                <div
                  class="bp3-input-group bp3-disabled"
                >
                  <input
                    autocomplete="off"
                    class="bp3-input"
                    disabled=""
                    name="secretManagerIdentifier"
                    placeholder="- Select -"
                    style="padding-right: 0px;"
                    type="text"
                    value="swaraj gcp"
                  />
                  <span
                    class="bp3-input-action"
                  >
                    <span
                      class="bp3-icon bp3-icon-chevron-down StyledProps--main StyledProps--padding-small"
                      icon="chevron-down"
                    >
                      <svg
                        data-icon="chevron-down"
                        height="14"
                        viewBox="0 0 16 16"
                        width="14"
                      >
                        <desc>
                          chevron-down
                        </desc>
                        <path
                          d="M12 5c-.28 0-.53.11-.71.29L8 8.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l4-4A1.003 1.003 0 0012 5z"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <span
          class="bp3-popover-wrapper hoverMsg"
        >
          <span
            class="bp3-popover-target hoverMsgTarget"
          >
            <div
              class="InputWithIdentifier--txtNameContainer TextInput--main"
            >
              <div
                class="Layout--horizontal Layout--layout-spacing-xsmall StyledProps--main InputWithIdentifier--txtIdContainer"
              >
                <div
                  class="Text--withDocsTooltip"
                  data-tooltip-id="universalInputNameWithId"
                >
                  <p
                    class="StyledProps--font StyledProps--main"
                  >
                    Id
                  </p>
                </div>
                <p
                  class="StyledProps--font StyledProps--main"
                >
                  :
                </p>
                <span
                  class="bp3-popover-wrapper"
                >
                  <span
                    class="bp3-popover-target"
                  >
                    <div
                      class="bp3-editable-text bp3-disabled InputWithIdentifier--idValue"
                    >
                      <span
                        class="bp3-editable-text-content"
                        style="height: 1px; line-height: 1px; min-width: 0;"
                      >
                        swarajtest2
                      </span>
                    </div>
                  </span>
                </span>
              </div>
              <div
                class="bp3-form-group bp3-disabled"
                data-id="name-1"
              >
                <label
                  class="bp3-label"
                  for="name"
                >
                  <span
                    class="Tooltip--acenter"
                    data-tooltip-id="createUpdateSecretForm_name"
                  >
                    secrets.labelSecretName
                  </span>
                   
                  <span
                    class="bp3-text-muted"
                  />
                </label>
                <div
                  class="bp3-form-content"
                >
                  <div
                    class="bp3-input-group bp3-disabled"
                  >
                    <input
                      autocomplete="off"
                      class="bp3-input"
                      disabled=""
                      name="name"
                      type="text"
                      value="swarajtest2"
                    />
                  </div>
                </div>
              </div>
            </div>
          </span>
        </span>
        <span
          class="bp3-popover-wrapper hoverMsg"
        >
          <span
            class="bp3-popover-target hoverMsgTarget"
          >
            <div
              class=""
            >
              <label
                class="bp3-control bp3-radio bp3-disabled bp3-inline"
              >
                <input
                  checked=""
                  disabled=""
                  name="Blueprint3.RadioGroup-0"
                  type="radio"
                  value="Inline"
                />
                <span
                  class="bp3-control-indicator"
                />
                secrets.secret.inlineSecret
              </label>
              <label
                class="bp3-control bp3-radio bp3-disabled bp3-inline"
              >
                <input
                  disabled=""
                  name="Blueprint3.RadioGroup-0"
                  type="radio"
                  value="Reference"
                />
                <span
                  class="bp3-control-indicator"
                />
                secrets.secret.referenceSecret
              </label>
            </div>
          </span>
        </span>
        <div
          class="bp3-form-group"
          data-id="value-2"
        >
          <label
            class="bp3-label"
            for="value"
          >
            <span
              class="Tooltip--acenter"
              data-tooltip-id="createUpdateSecretForm_value"
            >
              secrets.labelValue
            </span>
             
            <span
              class="bp3-text-muted"
            />
          </label>
          <div
            class="bp3-form-content"
          >
            <div
              class="bp3-input-group"
            >
              <input
                autocomplete="off"
                class="bp3-input"
                name="value"
                placeholder="encrypted"
                type="password"
                value=""
              />
            </div>
          </div>
        </div>
        <span
          class="bp3-popover-wrapper hoverMsg"
        >
          <span
            class="bp3-popover-target hoverMsgTarget"
          >
            <span
              class=""
              tabindex="0"
            >
              <div
                class="bp3-form-group bp3-disabled"
                data-id="configureRegions-3"
              >
                <div
                  class="bp3-form-content"
                >
                  <label
                    class="bp3-control bp3-checkbox bp3-disabled StyledProps--font Checkbox--checkbox StyledProps--main StyledProps--inline"
                  >
                    <input
                      checked=""
                      disabled=""
                      name="configureRegions"
                      type="checkbox"
                    />
                    <span
                      class="bp3-control-indicator"
                    />
                    <span
                      class="Tooltip--acenter FormikForm--checkBoxDocTooltipLabel"
                      data-tooltip-id="createUpdateSecretForm_configureRegions"
                    >
                      secrets.secret.configureRegion
                    </span>
                  </label>
                </div>
              </div>
              <div
                class="bp3-form-group bp3-disabled"
                data-id="regions-4"
              >
                <label
                  class="bp3-label"
                  for="regions"
                >
                  <span
                    class="Tooltip--acenter"
                    data-tooltip-id="createUpdateSecretForm_regions"
                  >
                    secrets.secret.region
                  </span>
                   
                  <span
                    class="bp3-text-muted"
                  />
                </label>
                <div
                  class="bp3-form-content"
                >
                  <div
                    class="bp3-popover-wrapper MultiSelect--main bp3-fill"
                  >
                    <div
                      class="bp3-popover-target"
                    >
                      <div
                        class=""
                      >
                        <div
                          class="bp3-input bp3-tag-input bp3-disabled bp3-multi-select"
                        >
                          <div
                            class="bp3-tag-input-values"
                          >
                            <span
                              class="bp3-tag MultiSelect--tag"
                              data-tag-index="0"
                            >
                              <span
                                class="bp3-text-overflow-ellipsis bp3-fill"
                              >
                                us-east1
                              </span>
                            </span>
                            <span
                              class="bp3-tag MultiSelect--tag"
                              data-tag-index="1"
                            >
                              <span
                                class="bp3-text-overflow-ellipsis bp3-fill"
                              >
                                us-east4
                              </span>
                            </span>
                            <input
                              autocomplete="off"
                              class="bp3-input-ghost bp3-multi-select-tag-input-input"
                              disabled=""
                              name="regions"
                              value=""
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </span>
          </span>
        </span>
        <div
          class="bp3-form-group"
          data-id="description-5"
        >
          <label
            class="bp3-label"
            for="description"
          >
            <span
              class="Tooltip--acenter"
              data-tooltip-id="createUpdateSecretForm_description"
            >
              description (optional)
            </span>
             
            <span
              class="bp3-text-muted"
            />
          </label>
          <div
            class="bp3-form-content"
          >
            <textarea
              autocomplete="off"
              class="bp3-input bp3-fill"
              maxlength="1024"
              name="description"
            />
          </div>
        </div>
        <div
          class="bp3-form-group"
          data-id="tags-6"
        >
          <label
            class="bp3-label"
            for="tags"
          >
            <span
              class="Tooltip--acenter"
              data-tooltip-id="createUpdateSecretForm_tags"
            >
              tagsLabel (optional)
            </span>
             
            <span
              class="bp3-text-muted"
            />
          </label>
          <div
            class="bp3-form-content"
          >
            <div
              class="bp3-input bp3-tag-input"
            >
              <div
                class="bp3-tag-input-values"
              >
                <input
                  class="bp3-input-ghost"
                  data-mentions="kv-tag-input-tags"
                  placeholder="Type and press enter to create a tag"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <button
          aria-label="save"
          class="bp3-button Button--button StyledProps--font StyledProps--main StyledProps--intent-primary StyledProps--margin-top-large Button--with-current-color Button--variation Button--variation-primary"
          type="submit"
        >
          <span
            class="bp3-button-text"
          >
            save
          </span>
        </button>
      </div>
    </form>
  </div>
</div>
`;
